src/rpdk/core/data_loaders.py

```diff
@@ -74,6 +74,33 @@ def make_resource_validator():
     return make_validator(schema)
 
 
+def make_resource_validator_with_additional_properties_check():
+    schema = resource_json(__name__, "data/schema/provider.definition.schema.v1.json")
+    dependencies = schema["definitions"]["validations"]["dependencies"]
+    properties_check = {
+        "properties": {
+            "$comment": "An object cannot have both defined and undefined \
+properties; therefore, patternProperties is not allowed when properties is specified.",
+            "not": {"required": ["patternProperties"]},
```

Comment:
is `not` supported?

Reply:
"`properties` and `patternProperties` it is not valid to use both properties and patternProperties together in the same shape, as a shape should not contain both defined and undefined values. In order to implement this, the set of undefined values should itself be a subshape."
From the ReadMe of [aws-cloudformation-resource-schema
](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema)

Reply:
Yeah discussed with @ammokhov. I will add couple more unit tests to test the case where providers set "additionalProperties" as true. We should not allow this as well.